### ISSUE TITLE
Feature: provide 20 minutes example data with holes

### ISSUE THREAD

**Issue by jerabaul29** [gauteh/concepts#2]
@gauteh it would be great if you can provide a 20 minutes example of data with holes - I am mostly interested in the vertical acceleration, but if you want to provide a netCDF files with more fields included it is fine for me of course :) . If it can be from a time when there is an interesting wave signal so that we can test stuff well that would be great :) .

In addition, if you can upload a 20 minutes example of data without holes at all (from a well functioning buoy with good GSM reception), that would be great, to be able to test the algorithms in a case where we know the answer :) .

**Comment by gauteh** [comment]
Two 20 minute segments, one with holes, one without. Same buoy.

[bug06.zip](https://github.com/gauteh/concepts/files/9983570/bug06.zip)

**Comment by jerabaul29** [comment]
Thanks for the data. I had a look at the data with segments, see https://github.com/jerabaul29/concepts/tree/feat/segments_fft . I think that it does not work as well as I had hoped, because the segments are really short, which means that the resolution of the FFT obtained based on individual segments and then averaging is really quite bad. I may be doing some other errors in addition too though, I have only had a short look so far.

**Comment by gauteh** [comment]
Ok, thanks for trying it out. Did you check with the nufft or with windowing? Here's an another attempt, but I don't trust the result - it is extremely sensitive to lower cutoff frequency. Do you know what the lower cutoff is for the OMB?

**Comment by gauteh** [comment]
https://github.com/gauteh/sfy-coastal-engineering/blob/main/data/2022-08-hustadvika-klakken/hs_spectra.py

**Comment by gauteh** [comment]
Probably also need to cutoff the first 128 samples of each segment, because of the FIR filter reset.

**Comment by gauteh** [comment]
And the spectrum is not so useful without integrating, I'm my case the buggy data blew up on integration (without filtering).

**Comment by jerabaul29** [comment]
I have only tried with the FFT on each segment.

The OMB typically can measure waves in the frequency range 0.05Hz - 1Hz, depending on the conditions, wave amplitude etc.

I usually never integrate the signal twice, that introduces a lot of noise; I rather take the FFT of the wave acceleration, and then multiply by 1/(2 pi f)**4 to get the elevation spectrum. Though, depending on the integration method used, it may be relatively equivalent.

**Comment by jerabaul29** [comment]
We can have a chat some day if you want, feel free to book yourself in my calendar (of course when it fits you, no hurry :) ).

**Comment by gauteh** [comment]
Yeah, I also integrate in the frequency domain. Let's talk about it soon 🙂

lør. 12. nov. 2022, 19:27 skrev JR ***@***.***>:

**Comment by jerabaul29** [comment]
Actually, I did a test on the data with no holes, creating synthetic holes, and the results taking the FFT with the full signal vs the artificially patchy signal agree quite well; I now wonder if the data with holes is not with quite little intrinsic signal anyways.

See:

- orange: the spectrum for the time series with no holes
- blue: the spectrum for the time series with no holes, creating holes artificially

![comparison_with_without_holes](https://user-images.githubusercontent.com/8382834/201491215-c6c3973b-3507-482f-b813-3269bf3ae655.png)

Pushed to: https://github.com/jerabaul29/concepts/tree/feat/segments_fft , at the end.

**Comment by jerabaul29** [comment]
(I hope there are no issues with what I have been doing in this notebook so far, a bit little time available so things are not as double checked as they could; but if things are mostly correct (which I think / hope), the FFT on each segment and averaging method seems actually quite ok despite the reduced resolution; I do not think I will have the time to do much more work myself on this, but I will be happy to discuss with you and "hand you in" the results / ideas I had a look at if you want to use them :) ).

**Comment by gauteh** [comment]
The gaps in the real data are not regular (can vary more than +/- 20 seconds), also the sample times do not align perfectly between the segments. Nor is the timing of the GPS good enough to capture that difference.
